src/main.rs

```diff
@@ -195,13 +195,13 @@ pub mod content {
                     }
 
                     let (screen_from, screen_to) = match (screens.get(from), screens.get(to)) {
-                        (Some(screen_from), Some(screen_to)) => (screen_from, screen_to),
+                        (Some(screen_from), Some(screen_to)) => (screen_from.clone(), screen_to.clone()),
                         _ => break,
                     };
 
                     screens.insert(
                         format!("{}-{}", to, index),
-                        generate_intermediate_screen(screen_from, screen_to, &word, &intermediate),
+                        generate_intermediate_screen(&screen_from, &screen_to, &word, &intermediate),
```

Comment:
Necessary due to the previous fix for the borrow warning.

Reply:
A better would be moving the result of `generate_intermediate_screen` to a local variable, no cloning is needed:
```
let generate_screen = generate_intermediate_screen(screen_from, screen_to, &word, &intermediate);
screens.insert(format!("{}-{}", to, index), generated_screen)
```

Reply:
Yes :100: I changed it to your solution.